src/create-with-theme.js

```diff
@@ -0,0 +1,39 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const channel = require('./channel');
+
+function createWithTheme(CHANNEL = channel) {
+  return Component =>
+    class withTheme extends React.Component {
+      static contextTypes = {
+        [CHANNEL]: PropTypes.func,
+      };
+
+      componentWillMount() {
+        if (!this.context[CHANNEL]) {
+          throw new Error(
```

Comment:
yeah but should this really break the UI? I s this a kind of error UI should stop working on?